projections.go

```diff
@@ -0,0 +1,244 @@
+package eventsourcing
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/hallgren/eventsourcing/core"
+)
+
+type fetchFunc func() (core.Iterator, error)
+type callbackFunc func(e Event) error
+
+type ProjectionHandler struct {
+	register *Register // used to map the event types
```

Comment:
I think that the comment should be part of the  `Register` type? 
```suggestion
	register *Register
```